### PR TITLE
Profile page and nav bar #12 -- Fixed

### PR DESCRIPTION
**Pull request by dipshojol**

### CODE REVIEW

**Review by aitchslash — APPROVED** [review]

**Review by aeciorc — CHANGES_REQUESTED** [review]
A few pointers:

1)
The dashboard route shouldn't actually be named (i.e, its path should be "/"). So the profile, message and discover routes would be accessed like this: "localhost:3000/profile", "localhost:3000/discover", "localhost:3000/messages". 

2)The profile route should be the default, so accessing "localhost:3000" would take them to "localhost:3000/profile" (assuming user is logged in, but you don't need to worry about that for now). You're probably gonna want to use a Switch component: https://github.com/ReactTraining/react-router/blob/master/packages/react-router/docs/api/Switch.md

3) I noticed that posts are only displayed if we're on the /follow route. That shouldn't be the case

4) Hitting the "message" and "follow" button shouldn't actually take you to any frontend route. That is, we don't need a "profile/follow" and "profile/message" route. 

5) The MakeAPICall file shouldn't be in the components folder. Move it somewhere else, maybe in the "utils" directory?

6) I'd rather JSS styling was used in this project instead of plain css. Take a look at how @Badr-KH styled the forms (utils/formstyles)

**Comment by dipshojol** [comment]
@aeciorc, Made the changes please check and also please let me know if you want me to make any other correction. for now, i am using css and slowly i am gonna convert them into JSS.

File: client/src/pages/Profile.js [hunk 1]
```diff
@@ -25,18 +22,12 @@ const Profile = () => {
                     </div>
                     <div className="follow-container">
                         <ul>
-                            <NavLink to="/dashboard/profile/follow" activeClassName="current"><li>Follow</li></NavLink>
-                            <NavLink to="/dashboard/profile/message" activeClassName="current"><li>Message</li></NavLink>
+                            <NavLink to="/profile/follow" activeClassName="current"><li>Follow</li></NavLink>
```
Comment (aeciorc) [inline client/src/pages/Profile.js]: These shouldn't be NavLinks. They should be regular buttons that will have an action in the future. So they shouldn't take the user anywhere

**Review by aeciorc — APPROVED** [review]
Just one more thing to fix (see my comments) and then you can merge

File: client/src/Components/Nav.js [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+                <h2>Instafy</h2>
+                <TextField
+                    id="outlined-basic"
+                    className={classes.textField}
+                    label="Share and enjoy music "
+                    margin="normal"
+                    variant="outlined"
+                    />
+                <ul className={classes.ul}>
+                    <NavLink className={classes.a} to="/profile" activeClassName="active"><li className={classes.li}>Share music</li></NavLink>
```
Comment (aeciorc) [inline client/src/Components/Nav.js:60]: Nice job with activeClassName, I didn't know about that prop. Makes things easier for sure!

**Comment by aeciorc** [comment]
Can you also make the navbar profile picture a link that points to "/profile" ? I've realized there's currently not link to profile

**Comment by dipshojol** [comment]
hi @aeciorc, please check

**Comment by aeciorc** [comment]
Feel free to merge

**Pull request merged**